titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/evacuation/AgentInstanceEvacuator.java

```diff
@@ -0,0 +1,118 @@
+package com.netflix.titus.supplementary.relocation.evacuation;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.netflix.titus.api.agent.model.AgentInstance;
+import com.netflix.titus.api.agent.model.AgentInstanceGroup;
+import com.netflix.titus.api.agent.model.InstanceOverrideState;
+import com.netflix.titus.common.runtime.TitusRuntime;
+import com.netflix.titus.runtime.connector.agent.AgentDataReplicator;
+import com.netflix.titus.runtime.connector.eviction.EvictionDataReplicator;
+import com.netflix.titus.runtime.connector.eviction.EvictionServiceClient;
+import com.netflix.titus.runtime.connector.jobmanager.JobDataReplicator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.Subscription;
+
+@Singleton
+public class AgentInstanceEvacuator {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentInstanceEvacuator.class);
+
+    private final static long STALENESS_THRESHOLD_MS = 5_000;
+
+    private final Subscription subscription;
+
+    private final AtomicReference<EvacuationExecutor> executorRef = new AtomicReference<>();
+
+    private final AgentDataReplicator agentDataReplicator;
+    private final JobDataReplicator jobDataReplicator;
+    private final EvictionDataReplicator evictionDataReplicator;
+    private final EvictionServiceClient evictionServiceClient;
+    private final EvacuationMetrics metrics;
+
+    @Inject
+    public AgentInstanceEvacuator(AgentDataReplicator agentDataReplicator,
+                                  JobDataReplicator jobDataReplicator,
+                                  EvictionDataReplicator evictionDataReplicator,
+                                  EvictionServiceClient evictionServiceClient,
+                                  TitusRuntime titusRuntime) {
+        this.agentDataReplicator = agentDataReplicator;
+        this.jobDataReplicator = jobDataReplicator;
+        this.evictionDataReplicator = evictionDataReplicator;
+        this.evictionServiceClient = evictionServiceClient;
+        this.metrics = new EvacuationMetrics(titusRuntime);
+
+        this.subscription = Observable.interval(0, 5, TimeUnit.SECONDS).subscribe(tick -> {
```

Comment:
Will this evolve someday to execute as many as possible within the disruption budget, instead of one agent every 5+ seconds?